src/web/browser/App.tsx

```diff
@@ -0,0 +1,199 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+import { Nav } from '@frontend/web/components/Nav/Nav';
+import { EditionDropdown } from '@frontend/web/components/EditionDropdown';
+import { MostViewedFooter } from '@frontend/web/components/MostViewed/MostViewedFooter/MostViewedFooter';
+import { MostViewedRightWrapper } from '@frontend/web/components/MostViewed/MostViewedRight/MostViewedRightWrapper';
+import { Counts } from '@frontend/web/components/Counts';
+import { RichLinkComponent } from '@frontend/web/components/elements/RichLinkComponent';
+import { ReaderRevenueLinks } from '@frontend/web/components/ReaderRevenueLinks';
+import { CMP } from '@frontend/web/components/CMP';
+import { Onwards } from '@frontend/web/components/Onwards/Onwards';
+import { SlotBodyEnd } from '@frontend/web/components/SlotBodyEnd';
+import { SubNav } from '@frontend/web/components/SubNav/SubNav';
+
+type Props = { CAPI: CAPIType; NAV: NavType };
+
+type RootType =
+    | 'reader-revenue-links-header'
+    | 'nav-root'
+    | 'sub-nav-root'
+    | 'edition-root'
+    | 'most-viewed-right'
+    | 'share-comment-counts'
+    | 'most-viewed-footer'
+    | 'reader-revenue-links-footer'
+    | 'slot-body-end'
+    | 'cmp'
+    | 'onwards-content'
+    | 'rich-link';
+
+export const hydrateApp = ({ CAPI, NAV }: { CAPI: CAPIType; NAV: NavType }) => {
+    ReactDOM.render(
+        <App CAPI={CAPI} NAV={NAV} />,
+        document.getElementById('react-root'),
+    );
+};
+
+const App = ({ CAPI, NAV }: Props) => {
+    const richLinks: {
+        element: RichLinkBlockElement;
+        root: RootType;
+        richLinkIndex: number;
+    }[] = [];
+    CAPI.blocks[0].elements.map((element, i) => {
+        if (
+            element._type ===
+            'model.dotcomrendering.pageElements.RichLinkBlockElement'
+        ) {
+            richLinks.push({
+                element,
+                root: `rich-link`,
```

Comment:
It is expected to have 'i' appended, but I went with using a special prop (`richLinkIndex`) and appending it later so that I can preserve the typing of this field.